loader/include/Geode/ui/PaddingNode.hpp

```diff
@@ -0,0 +1,133 @@
+#pragma once
+
+#include <cocos2d.h>
+
+namespace geode {
+    class GEODE_DLL PaddingNode : public cocos2d::CCNode {
+    public:
+        enum class PaddingType {
+            // The container will have its size adjusted to fit the padding
+            Contained,
+            // This node will have its size adjusted to fit the padding
+            Uncontained
+        };
+
+        struct Padding {
+            operator float() const;
+            operator cocos2d::CCPoint() const;
+
+            float top;
+            float right;
+            float bottom;
+            float left;
+
+            Padding();
+            Padding(cocos2d::CCPoint padding);
```

Comment:
`CCPoint` is not pointer-size on all platforms (Android32) so this should be `const&`